app/models/manageiq/providers/ansible_tower/shared/automation_manager/credential.rb

```diff
@@ -33,6 +33,10 @@ def provider_object(connection = nil)
     (connection || connection_source.connect).api.credentials.find(manager_ref)
   end
 
+  def manager_ref
+    self[:manager_ref].to_i
```

Comment:
`super.to_i` would be cleaner

Reply:
I don't have a preference but the `self[:attr]` is based on https://stackoverflow.com/questions/21835116/how-can-i-overwrite-a-getter-method-in-an-activerecord-model

Reply:
If we want to use  something like `manager_ref_obj` can that be a virtual column?  This would still requires us to modify all the callers.

Reply:
@bdunne  overriding `manager_ref` which was my initial approach.  See @agrare 's concern in this thread

Reply:
I see.  I guess my concern is that all callers need to be updated and new code needs to know about the new method.

Reply:
yeah, that's the drawback of creating a new method.